### ISSUE TITLE
Wrong Composer on WSL with interop enabled

### ISSUE THREAD

**Issue by Rainrider** [symfony-cli/symfony-cli#43]
The cli picks up the wrong composer when run on Windows Subsystem for Linux with interop enabled. In interop mode (on by default) the Windows path is appended to the Linux one (unless `appendWindowsPath` is set to `false` in `/etc/wsl.conf`). This assures that Linux binaries are preferred to Windows ones, however this seems to not be honored by Symfony's CLI.

Linux (WSL)

```console
$ which symfony
/usr/local/bin/symfony
$ symfony version
Symfony CLI version 5.0.9 (c) 2017-2022 Symfony SAS (2022-01-11T15:42:39Z - stable)
$ which composer
/usr/local/bin/composer
$ composer --version
Composer version 2.2.4 2022-01-08 12:30:42
$ symfony composer --version
PHP Deprecated:  Return type of Symfony\Component\Console\Helper\HelperSet::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///mnt/c/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Helper/HelperSet.php on line 112

Deprecated: Return type of Symfony\Component\Console\Helper\HelperSet::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///mnt/c/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Helper/HelperSet.php on line 112
Composer version 2.1.3 2021-06-09 16:31:20
```

Path output truncated for brevity (order preserved)

```console
$ echo $PATH | tr ':' '\n'
/usr/local/sbin
/usr/local/bin
/usr/sbin
/usr/bin
/sbin
/bin
/usr/games
/usr/local/games
/usr/lib/wsl/lib
/mnt/c/Windows/System32
/mnt/c/Windows
/mnt/c/ProgramData/ComposerSetup/bin
```

Windows (PowerShell)

```console
$ echo $(get-command composer).Source
C:\ProgramData\ComposerSetup\bin\composer.bat
$ composer --version
Composer version 2.1.3 2021-06-09 16:31:20
```

If I disable Windows interop in WSL, Symfony detects the proper composer.

**Comment by fabpot** [comment]
Can you investigate what is happening and see if you can propose a fix? Maybe a duplicate of #50?

**Comment by vbrouet** [comment]
Had the same issue. I did a bit of investigation.

Root cause is because `composer.phar`  in the windows path takes precedence over `composer`

This is the affected code:
https://github.com/symfony-cli/symfony-cli/blob/00cbca0c5f0d8a5240a76287fc376070b5e752e3/local/php/composer.go#L134-L141

The order of the binaries is `composer2` (default value of extraBin) > `composer.phar` > `composer`

It may be more inuitive use `extraBin` > `composer` > `composer.phar`?

**Comment by fabpot** [comment]
Makes sense, fixed in bd06f2f76aa3dc914d53fc1d4d8492a413b7548a
